server/player.lua

```diff
@@ -109,17 +129,46 @@ exports('SetPlayerPrimaryJob', SetPlayerPrimaryJob)
 ---@param citizenid string
 ---@param jobName string
 ---@param grade integer
+---@return boolean success
+---@return ErrorResult? errorResult
 function AddPlayerToJob(citizenid, jobName, grade)
     -- unemployed job is the default, so players cannot be added to it
-    if jobName == 'unemployed' then return end
+    if jobName == 'unemployed' then
+        return false, {
+            code = 'unemployed',
+            message = 'players cannot be added to the unemployed job'
+        }
+    end
     local job = GetJob(jobName)
-    assert(job ~= nil, 'job not found: ' .. jobName)
-    assert(job.grades[grade] ~= nil, string.format('job %s does not have grade %s', jobName, grade))
+    if not job then
+        return false, {
+            code = 'job_not_found',
+            message = jobName .. ' does not exist in core memory'
+        }
+    end
+    if not job.grades[grade] then
+        return false, {
+            code = 'job_missing_grade',
+            message = string.format('job %s does not have grade %s', jobName, grade),
+        }
+    end
 
     local player = GetPlayerByCitizenId(citizenid) or GetOfflinePlayer(citizenid)
-    assert(player ~= nil, string.format('player not found with citizenid %s', citizenid))
-    if player.PlayerData.jobs[jobName] == grade then return end
-    assert(qbx.table.size(player.PlayerData.jobs) < maxJobsPerPlayer or player.PlayerData.jobs[jobName], 'player already has maximum amount of jobs allowed')
+    if not player then
+        return false, {
+            code = 'player_not_found',
+            message = string.format('player not found with citizenid %s', citizenid)
+        }
+    end
+    if player.PlayerData.jobs[jobName] == grade then
+        return true
+    end
+    if qbx.table.size(player.PlayerData.jobs) >= maxJobsPerPlayer and not player.PlayerData.jobs[jobName] then
```

Comment:
question, why not add spaces between if statements? I personally don't like it when they're grouped together as they're meant to do different things and may make it difficult to read.

Reply:
added more whitespace.